tests/rptest/utils/si_utils.py

```diff
@@ -288,15 +290,22 @@ def __init__(self, brokers, name, logger, timeout_sec: float = 60.0):
         self.reconnect()
 
     def reconnect(self):
-        self.producer = confluent_kafka.Producer({
-            'bootstrap.servers':
-            self.brokers,
-            'transactional.id':
-            self.name,
-            'transaction.timeout.ms':
-            5000,
-        })
-        self.producer.init_transactions(self.timeout_sec)
+        def init():
+            try:
+                self.producer = ck.Producer({
+                    'bootstrap.servers': self.brokers,
+                    'transactional.id': self.name,
+                    'transaction.timeout.ms': 5000,
+                })
+                self.producer.init_transactions(self.timeout_sec)
+                return True
+            except ck.cimpl.KafkaException as e:
+                kafka_error = e.args[0]
+                if kafka_error.code() == ck.cimpl.KafkaError.NOT_COORDINATOR:
+                    return False
+                raise
+
+        wait_until(init, timeout_sec=10, backoff_sec=1)
```

Comment:
I think that the producer uses 60s timeout which is passed to the init_transactions method, so 10s timeout doesn't make sense (or timeout is not handled correctly inside the confluent kafka python package or librdkafka which it uses internally)

Reply:
Log sample

```
[DEBUG - 2022-09-07 12:46:38,559 - kafka_cli_tools - _execute - lineno:267]: Created topic topic-uoxzdbzcgp.
[INFO  - 2022-09-07 12:46:38,559 - runner_client - log - lineno:278]: RunnerClient: rptest.tests.e2e_topic_recovery_test.EndToEndTopicRecovery.test_restore_with_aborted_tx.recovery_overrides=: Running...
[ERROR - 2022-09-07 12:46:43,561 - cluster - wrapped - lineno:37]: Test failed, doing failure checks...
Traceback (most recent call last):
  File "/root/tests/rptest/services/cluster.py", line 35, in wrapped
    r = f(self, *args, **kwargs)
  File "/root/tests/rptest/tests/e2e_topic_recovery_test.py", line 212, in test_restore_with_aborted_tx
    producer = Producer(self.redpanda.brokers(), "topic-recovery-tx-test",
  File "/root/tests/rptest/utils/si_utils.py", line 287, in __init__
    self.reconnect()
  File "/root/tests/rptest/utils/si_utils.py", line 298, in reconnect
    self.producer.init_transactions()
cimpl.KafkaException: KafkaError{code=NOT_COORDINATOR,val=16,str="Failed to initialize Producer ID: Broker: Not coordinator"}
[INFO  - 2022-09-07 12:46:43,562 - redpanda - decode_backtraces - lineno:1339]: Decoding backtraces on docker-rp-18.
[DEBUG - 2022-09-07 12:46:43,562 - remoteaccount - _log - lineno:160]: root@docker-rp-18: Running ssh command: /opt/scripts/seastar-addr2line -e /var/lib/buildkite-agent/builds/buildkite-amd64-xfs-builders-i-0c8e22cb956c65365-1/redpanda/redpanda/vbuild/release/clang/dist/local/redpanda/libexec/redpanda -f /var/lib/redpanda/redpanda.log > /var/lib/redpanda/redpanda_backtrace.log 2>&1 && find /var/lib/redpanda/redpanda_backtrace.log -type f -size 0 -delete
```

The test started at 12:46:38,559 and failed at 12:46:43,561 so it lasted at most 8s (not 60s) and 10s looks fine. Let's merge it and iterate if it doesn't work.

Reply:
> I think that the producer uses 60s timeout which is passed to the init_transactions method, so 10s timeout doesn't make sense

Thats a good point, didn't realize default is 60s. Don't have a strong preference here since NOT_COORDINATOR (intent of retries in this PR) is usually super quick to return and not timeout.